src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -159,7 +159,7 @@ public static IResourceBuilder<T> WithEnvironment<T>(this IResourceBuilder<T> bu
     /// <param name="builder">The resource builder.</param>
     /// <param name="args">The arguments to be passed to the container when it is started.</param>
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<T> WithArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : IResourceWithArgs
+    public static IResourceBuilder<T> WithArgs<T>(this IResourceBuilder<T> builder, params object[] args) where T : IResourceWithArgs
```

Comment:
This is an API breaking change. We can't change existing APIs, we can only add new ones.